demos/projects/ST/b-l475e-iot01a/port/sockets_wrapper_stm32l475.c

```diff
@@ -0,0 +1,653 @@
+                xRetVal = SOCKETS_SOCKET_ERROR;
+                break;
+            }
+        }
+        else
+        {
+            /* Semaphore wait time was longer than the receive timeout so this
+             * is also a socket timeout. Returning SOCKETS_EWOULDBLOCK will
+             * cause mBedTLS to fail and so we must return zero.*/
+            xRetVal = 0;
+            break;
+        }
+    }
+
+    /* The following code attempts to revive the Inventek WiFi module
+     * from its unusable state.*/
+    if( xWiFiResult == WIFI_STATUS_ERROR )
+    {
+        /* Reset the WiFi Module. Since the WIFI_Reset function
+         * acquires the same semaphore, we must not acquire
+         * it. */
+        if( WIFI_ResetModule() == WIFI_STATUS_OK )
+        {
+            /* Try to acquire the semaphore. */
+            if( xSemaphoreTake( xWifiSemaphoreHandle, portMAX_DELAY ) == pdTRUE )
+            {
+                /* Reinitialize the socket structures which
+                 * marks all sockets as closed and free. */
+                SOCKETS_Init();
+
+                /* Return the semaphore. */
+                ( void ) xSemaphoreGive( xWifiSemaphoreHandle );
+            }
+
+            /* Set the error code to indicate that
+             * WiFi needs to be reconnected to network. */
+            xRetVal = SOCKETS_PERIPHERAL_RESET;
+        }
+    }
+
+    return xRetVal;
+}
+/*-----------------------------------------------------------*/
+
+BaseType_t Sockets_Send( SocketHandle xSocket,
+                         const uint8_t * pucData,
+                         size_t xDataLength )
+{
+    uint32_t ulSocketNumber = ( uint32_t ) xSocket;
+    STSecureSocket_t * pxSecureSocket;
+    uint16_t usSentBytes = 0;
+    BaseType_t xRetVal = SOCKETS_SOCKET_ERROR;
+    WIFI_Status_t xWiFiResult = WIFI_STATUS_OK;
+
+    /* Shortcut for easy access. */
+    pxSecureSocket = &( xSockets[ ulSocketNumber ] );
+
+    /* Try to acquire the semaphore. */
+    if( xSemaphoreTake( xWifiSemaphoreHandle, xSemaphoreWaitTicks ) == pdTRUE )
+    {
+        // /* Since WiFi module has only one timeout, this needs
+        //  * to be set per send and receive operation to the
+        //  * respective send or receive timeout. Also, this
+        //  * must be done after acquiring the semaphore as the
+        //  * xWiFiModule is a shared object.*/
+        // if( pxSecureSocket->ulSendTimeout == 0 )
+        // {
+        //     /* Set the SPI timeout to the maximum uint32_t value.
+        //      * This is a little over 49 days. */
+        //     xWiFiModule.xWifiObject.Timeout = 0xFFFFFFFF;
+        // }
+        // else
+        // {
+        //     /* The maximum timeout for Inventek module is 30 seconds.
+        //      * This timeout is about 65 seconds, so the module should
+        //      * timeout before the SPI. */
+        //     xWiFiModule.xWifiObject.Timeout = ES_WIFI_TIMEOUT;
+        // }
```

Comment:
curious why this is commented out (might not have been by you)

Reply:
removed